File: go/arrays-and-hashing/two_sum.go
package arrays_and_hashing

func TwoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]

		if index, exists := numMap[complement]; exists {
			return []int{index, i}
		}

		numMap[nums[i]] = i
	}

	return []int{}
}

File: swift/Sources/LeetCode/ArraysAndHashing/TwoSum.swift
public class TwoSum {
    public init() {}

    public func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numMap: [Int: Int] = [:]

        for i in 0..<nums.count {
            let complement = target - nums[i]

            if let index = numMap[complement] {
                return [index, i]
            }

            numMap[nums[i]] = i
        }

        return []
    }
}